src/scr.c

```diff
@@ -1782,6 +1783,95 @@ int SCR_Init()
     return SCR_FAILURE;
   }
 
+  /* coonfigure used libraries */
+  {
+    kvtree* axl_config = kvtree_new();
+    assert(axl_config);
+
+    if (kvtree_util_set_double(axl_config, AXL_KEY_CONFIG_FLUSH_ASYNC_BW,
+                               scr_flush_async_bw) != KVTREE_SUCCESS) {
+      scr_abort(-1, "Failed to set AXL config option %s @ %s:%d",
+        AXL_KEY_CONFIG_FLUSH_ASYNC_BW, __FILE__, __LINE__
+      );
+    }
+    if (kvtree_util_set_int(axl_config, AXL_KEY_CONFIG_FLUSH_ASYNC_PERCENT,
+                            scr_flush_async_percent) != KVTREE_SUCCESS) {
+      scr_abort(-1, "Failed to set AXL config option %s @ %s:%d",
+        AXL_KEY_CONFIG_FLUSH_ASYNC_PERCENT, __FILE__, __LINE__
+      );
+    }
+    if (kvtree_util_set_bytecount(axl_config, AXL_KEY_CONFIG_FILE_BUF_SIZE,
+                                  scr_file_buf_size) != KVTREE_SUCCESS) {
+      scr_abort(-1, "Failed to set AXL config option %s @ %s:%d",
+        AXL_KEY_CONFIG_FILE_BUF_SIZE, __FILE__, __LINE__
+      );
+    }
+    if (kvtree_util_set_int(axl_config, AXL_KEY_CONFIG_DEBUG,
+                            scr_debug) != KVTREE_SUCCESS) {
+      scr_abort(-1, "Failed to set AXL config option %s @ %s:%d",
+        AXL_KEY_CONFIG_DEBUG, __FILE__, __LINE__
+      );
+    }
+    if (kvtree_util_set_int(axl_config, AXL_KEY_CONFIG_MKDIR,
+                            scr_debug) != KVTREE_SUCCESS) {
```

Comment:
I don't think we want to use scr_debug here, but I don't know if we have a variable.  Anyway, currently SCR creates directories, so we can always configure AXL to skip that, so we could replace ``scr_debug`` with ``0``.

Reply:
huh, sorry. This should not be `scr_debug` but something like `scr_make_directories` to match AXL's internal variable. Sorry.

Fixed in 1ffd9e4 which introduces a new SCR option `SCR_MKDIR` to set `AXL_MKDIR`.

Reply:
Yep, that's good.  This will be a nice option on systems where SCR can offload directory creation to something else.  There is a wrinkle here, though.

I think we want to flip directory creation with this option.  If SCR creates the directories, then we want SCR to configure AXL to not also create directories.  On the other hand, if the user configures SCR to *not* create directories, SCR should tell AXL that it has to.

Can you also open an issue so we remember to update the code to disable the mkdir part of SCR if someone sets this value?

Reply:
Uhm right now SCR does not actually do anything with the parameter other than passing it to AXL. It's comment does indicate that SCR may do the directory creation though that is mostly to not expose SCR's internals (namely that AXL exists) to a user code.